File: OperaWebsite/modules/activity/src/main/java/com/opera/activity/domain/Activity.java
package com.opera.activity.domain;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;

import java.io.Serial;
import java.io.Serializable;
import java.util.Date;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * 活动表
 */
@Data
@AllArgsConstructor
@NoArgsConstructor
@TableName(value = "activity")
public class Activity implements Serializable {
    /**
     * 活动编号
     */
    @TableId(value = "activity_id", type = IdType.ASSIGN_ID)
    private Long activityId;

    /**
     * 活动名称
     */
    @TableField(value = "activity_name")
    private String activityName;

    /**
     * 活动简介
     */
    @TableField(value = "activity_info")
    private String activityInfo;

    /**
     * 活动地址
     */
    @TableField(value = "activity_address")
    private String activityAddress;

    /**
     * 活动发起人
     */
    @TableField(value = "playgoer_id")
    private Long playgoerId;

    /**
     * 活动人数
     */
    @TableField(value = "activity_num")
    private Integer activityNum;

    /**
     * 活动开始时间
     */
    @TableField(value = "start_time")
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm", timezone = "GMT+8")
    private Date startTime;

    /**
     * 活动结束时间
     */
    @TableField(value = "end_time")
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm", timezone = "GMT+8")
    private Date endTime;

    /**
     * 是否删除
     */
    @TableField(value = "is_delete")
    private Integer isDelete;

    /**
     * 创建时间
     */
    @TableField(value = "created_at")
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss", timezone = "GMT+8")
    private Date createdAt;

    /**
     * 更新时间
     */
    @TableField(value = "updated_at")
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss", timezone = "GMT+8")
    private Date updatedAt;

    /**
     * 创建人
     */
    @TableField(value = "created_by")
    private Long createdBy;

    /**
     * 修改人
     */
    @TableField(value = "updated_by")
    private Long updatedBy;

    @Serial
    private static final long serialVersionUID = 1L;
}

File: OperaWebsite-UI/src/api/types/activity.ts
export interface Activity {
	/**
	 * 活动编号
	 */
	activityId?: string
	/**
	 * 活动名称
	 */
	activityName?: string
	/**
	 * 活动简介
	 */
	activityInfo?: string
	/**
	 * 活动地址
	 */
	activityAddress?: string
	/**
	 * 活动发起人
	 */
	playgoerId?: string
	/**
	 * 活动人数
	 */
	activityNum?: number
	/**
	 * 活动开始时间
	 */
	startTime?: Date
	/**
	 * 活动结束时间
	 */
	endTime?: Date
	/**
	 * 是否删除
	 */
	isDelete?: number
	/**
	 * 创建时间
	 */
	createdAt?: Date
	/**
	 * 更新时间
	 */
	updatedAt?: Date
	/**
	 * 创建人
	 */
	createdBy?: string
	/**
	 * 修改人
	 */
	updatedBy?: string
}
export interface ActivityVO {
	/**
	 * 发布者编号
	 */
	playgoerId?: string
	/**
	 * 发布者头像
	 */
	playgoerAvatar?: string
	/**
	 * 发布者姓名
	 */
	playgoerName?: string
	/**
	 * 发布者签名
	 */
	playgoerInfo?: string
	/**
	 * 活动名称
	 */
	activityName?: string
	/**
	 * 活动简介
	 */
	activityInfo?: string
	/**
	 * 活动地址
	 */
	activityAddress?: string
	/**
	 * 活动剩余名额
	 */
	activityQuota?: number
	/**
	 * 活动人数
	 */
	activityNum?: number
	/**
	 * 活动开始时间
	 */
	startTime?: Date
	/**
	 * 活动结束时间
	 */
	endTime?: Date
	/**
	 * 创建时间
	 */
	createdAt?: Date
	/**
	 * 更新时间
	 */
	updatedAt?: Date
}
